pkgs/development/python-modules/mozjpeg_lossless_optimization/default.nix

```diff
@@ -0,0 +1,34 @@
+{
+  lib,
+  python3Packages,
+  fetchPypi,
+  mailman,
+  nixosTests,
+  buildPythonPackage,
+  mozjpeg,
+  cmake,
+}:
+buildPythonPackage rec {
+  pname = "mozjpeg_lossless_optimization";
+  version = "1.1.3";
+
+  src = fetchPypi {
+    inherit version;
+    pname = "mozjpeg-lossless-optimization";
+    hash = "sha256-cl2Ydy6UP8oYsIAcuU5kXEd/9S5WrQsnvdt23fCRyj4=";
+  };
+
+  # This package needs cmake, but it is not the default builder
+  dontUseCmakeConfigure = true;
+
+  buildInputs = [ mozjpeg ];
+  nativeBuildInputs = [ cmake ];
+  propagatedBuildInputs = [ python3Packages.cffi ];
+
```

Comment:
```suggestion
  nativeCheckInputs = [ pytestCheckHook ];
```
For some reason, this fails with an error:
```
==================================== ERRORS ====================================
_________________ ERROR collecting tests/test_mozjpeg_opti.py __________________
ImportError while importing test module '/build/mozjpeg-lossless-optimization-1.1.3/tests/test_mozjpeg_opti.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/cfvxmzfd7fsndmcdfg6hdyhj5ag9xhcw-python3-3.12.7/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_mozjpeg_opti.py:3: in <module>
    import mozjpeg_lossless_optimization
mozjpeg_lossless_optimization/__init__.py:1: in <module>
    from .mozjpeg_opti import optimize
mozjpeg_lossless_optimization/mozjpeg_opti.py:1: in <module>
    from ._mozjpeg_opti import lib, ffi
E   ModuleNotFoundError: No module named 'mozjpeg_lossless_optimization._mozjpeg_opti'
=========================== short test summary info ============================
ERROR tests/test_mozjpeg_opti.py
```
The module imports fine in a normal Python environment though.

Reply:
Thank you! Activating the tests was the right approach.

- To resolve the issue, I removed the directory that shared the module's name. It appears to be a known problem: https://github.com/NixOS/nixpkgs/issues/255262.
 - I switched fetchPyPi to fetchFromGithub because the PyPI archive lacks a required file (image.jpg) needed for the tests.